agent/command/test_selection_get.go

```diff
@@ -153,14 +144,14 @@ func (c *testSelectionGet) Execute(ctx context.Context, comm client.Communicator
 	if c.Strategies != "" {
 		trimmedStrategies := strings.TrimSpace(c.Strategies)
 		request.Strategies = strings.Split(trimmedStrategies, ",")
+		trace.SpanFromContext(ctx).SetAttributes(attribute.StringSlice(testSelectionStrategiesAttribute, request.Strategies))
 	}
 
 	startTime := time.Now()
 	selectedTests, err := comm.SelectTests(ctx, conf.TaskData(), request)
 	durationMs := time.Since(startTime).Milliseconds()
 	trace.SpanFromContext(ctx).SetAttributes(attribute.Int64(testSelectionDurationMsAttribute, durationMs))
-
-	calledAPI = true // nolint:ineffassign // Used in defer.
+	trace.SpanFromContext(ctx).SetAttributes(attribute.Bool(testSelectionCalledAttribute, true))
```

Comment:
We should still explicitly set this to false when it's not called so that users can just check true/false rather than true vs. nonexistent field (which is harder).

One way to get the original defer to work as intended would be like this:
```go
defer func() {
    trace.SpanFromContext(ctx).SetAttributes(attribute.Bool(testSelectionCalledAttribute, calledAPI))
}()
```
The function arguments to the deferred function are resolved on the line where defer is declared (which was the original issue), but when the function actually executes, it can access the current value of the `calledAPI` bool because of the closure.